File: src/main/java/ru/mrak/service/dto/gallery/GalleryDto.java
package ru.mrak.service.dto.gallery;

import lombok.Builder;
import lombok.Data;
import lombok.experimental.Accessors;
import ru.mrak.model.enumeration.GalleryTypeEnum;

import java.util.ArrayList;
import java.util.List;

@Data
@Accessors(chain = true)
public class GalleryDto {
    private List<GalleryItemDto> galleryItems = new ArrayList<>();
    private GalleryTypeEnum type;
    private String title;
}

File: src/main/webapp/src/model/gallery/galleryDto.ts
import { GalleryItemDto } from '@/model/gallery/galleryItemDto'

export interface Gallery {
  galleryItems?: GalleryItemDto[]
  type?: string
  title?: string
}

export class GalleryDto implements Gallery {
  constructor (
    public galleryItems: GalleryItemDto[],
    public type?: string,
    public title?: string
  ) {
  }
}
